src/regdesk/communication_manager.js

```diff
@@ -107,22 +107,32 @@ export class CommunicationManager {
 
   /**
    * Handle messages from other extension components.
-   * @param {*} o - The request object.
+   * @param {object} o - The request object.
    * @param {string} o.type - The request type.
    * @param {*} o.payload - The request payload.
-   * @param {*} sender - The sender of the event.
+   * @param {chrome.runtime.MessageSender} sender - The sender of the event.
    * @param {function(*)} callback - Callback message to response to the event sender.
    * @return {boolean} - True to indicate this will response async.
    */
   #handleExtensionMessage({ type, payload }, sender, callback) {
     console.debug('Received extension message', type, payload, sender);
 
+    // NOTE: Any object that is passed through the message system will STRIP all
```

Comment:
The only methods are getters, eg `hasBeenCheckedIn` why not make those variables set by the constructor? Then we don't need to have this big clumsy copy going on.

Reply:
Therein lies the fun: It's recursive. The methods on the label were also removed, which meant the label that is generated wasn't able to print. Originally the getter for the label was a method called `getLabel()` as I don't really like getters that call constructors. The workaround didn't work :/

Rather than go down the rabbit hole of why this is happening I made a judgement call on limiting my deep knowledge of Chrome/JS internals and just made a copy constructor. It's absolutely the brute force solution but it's also the most reliable.

Reply:
You and I have different definitions of reliable :P